### PR TITLE
Feat/measurements parser

### PR DESCRIPTION
**Pull request by dnadales**

### CODE REVIEW

File: src/Cobra/ResultsParser.hs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+metricNameP = MetricName . T.pack <$> strP
+
+metricValuesP :: Parser MetricValues
+metricValuesP = (MetricValues . Map.fromList) <$> metricsP
+
+testResultP :: Parser (TestName, MetricValues)
+testResultP = (,) <$> testNameP <*> metricValuesP
+
+testNameP  :: Parser TestName
+testNameP = TestName . T.pack <$> strP
```
Comment (keremispirli) [inline src/Cobra/ResultsParser.hs]: Won't it be better if we order these by usage, like:
parseBOL (_what is BOL?_)
testResultP (_P for Parser?_)
testNameP
strP
lexer
metricValuesP
metricsP
measurementP
metricNameP
floatP
signP

File: test/Cobra/DataGen.hs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```
Comment (keremispirli) [inline test/Cobra/DataGen.hs]: Can't we name these with full words, like "DataGenerator"?

File: test/Cobra/ResultsParserSpec.hs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import           Test.QuickCheck
+
+    
+import           Cobra.ResultsParser
+import           Cobra.Data
+import           Cobra.DataGen ()
+
+asTestResult :: TestName -> MetricValues -> Text
+asTestResult (TestName tn) (MetricValues mnVals) =
+    quote tn <> " " <> T.concat (map mvAsResult (Map.toList  mnVals))
```
Comment (keremispirli) [inline test/Cobra/ResultsParserSpec.hs]: mnVals -> mVals ?

Comment (capitanbatata) [reply test/Cobra/ResultsParserSpec.hs]: Yes, makes sense.

File: src/Cobra/ResultsParser.hs (hunk 1 above)
Comment (capitanbatata) [reply src/Cobra/ResultsParser.hs]: BOL = Benchmark Output Line, I didn't want to give such a long name.

I agree with the order you propose.

File: test/Cobra/DataGen.hs (hunk 2 above)
Comment (capitanbatata) [reply test/Cobra/DataGen.hs]: In quick check the generators are usually suffixed with Gen. Hence the name.

**Comment by capitanbatata** [comment]
Feel free to make the changes above, and merge this.

**Review by keremispirli — APPROVED** [review]
Needs extra-deps in stack.yaml for building in Windows, but I think it's not a requirement now.

**Comment by dnadales** [comment]
We should make an issue for the extra deps needed in Windows, so that we can build on this platform as well.

**Pull request merged**